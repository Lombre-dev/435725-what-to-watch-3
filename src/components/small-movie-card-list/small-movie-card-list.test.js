import * as React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SmallMovieCardList from './small-movie-card-list';

const GENRES = [`Drama`, `Comedy`, `Kids & Family`];
const CURRENT_GENRE = GENRES[0];
const MOVIES = [
  {
    id: 0,
    title: `The Grand Budapest Hotel`,
    genres: [`Drama`],
    year: 2014,
    poster: `/img/the-grand-budapest-hotel-poster.jpg`,
    frames: [`/img/the-grand-budapest-hotel-poster.jpg`],
    ratingScore: 8.9,
    ratingReviewsCount: 240,
    preview: `/samples/sintel_trailer-480p.mp4`,
    src: `/samples/sintel_trailer-480p.mp4`,
    description: `Description`,
    story: `Story`,
    director: `Wes Andreson`,
    duration: 99,
    actors: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Some Actor 1`,
      `Some Actor 2`,
    ],
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    genres: [`Comedy`],
    year: 2014,
    poster: `/img/bohemian-rhapsody.jpg`,
    frames: [`/img/bohemian-rhapsody.jpg`],
    duration: 120,
    preview: ``,
    src: ``,
    ratingScore: 7.5,
    ratingReviewsCount: 100,
    description: `Description`,
    story: `Story`,
    director: `Director`,
    actors: [
      `Some Actor 1`,
      `Some Actor 2`,
      `Some Actor 3`,
      `Some Actor 4`,
      `Some Actor 5`,
    ],
  },
  {
    id: 2,
    title: `Macbeth`,
    genres: [`Kids & Family`],
    year: 2014,
    preview: ``,
    src: ``,
    poster: `/img/macbeth.jpg`,
    frames: [`/img/macbeth.jpg`],
    duration: 100,
    ratingScore: 6.9,
    ratingReviewsCount: 50,
    description: `Description`,
    story: `Story`,
    director: `Director`,
    actors: [
      `Some Actor 1`,
      `Some Actor 2`,
      `Some Actor 3`,
      `Some Actor 4`,
      `Some Actor 5`,
    ],
  },
];
const ACTIVE_ITEM_ID = -1;
const HAS_MORE_MOVIES = true;
const HANDLE_EVENT = () => {};

const mockStore = configureStore([]);

describe(`<SmallMovieCardList />`, () => {

  it(`render should be match markup`, () => {

    const store = mockStore({
      currentMovie: null,
      promoMovie: MOVIES[0],
      catalogGenres: GENRES,
      catalogGenre: CURRENT_GENRE,
      catalogMovies: MOVIES,
      hasMoreCatalogMovies: HAS_MORE_MOVIES,
    });

    const result = renderer
      .create(<Provider store={store}>
        <BrowserRouter>
          <SmallMovieCardList
            movies={MOVIES}
            activeItemId={ACTIVE_ITEM_ID}
            onItemHover={HANDLE_EVENT}
            onItemLeave={HANDLE_EVENT}
          />
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
