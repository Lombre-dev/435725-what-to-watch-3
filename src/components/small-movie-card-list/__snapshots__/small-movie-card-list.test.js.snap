// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallMovieCardList /> render should be match markup 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwal"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwal
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        src="img/bohemian-rhapsody.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="img/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
</div>
`;
